### PR TITLE
Integration test for secure nodeport service

### PR DESCRIPTION
**Pull request by anpanigr**
Integration test for following secure nodeport service  scenarios. 
 
1. Verifies  a `default-admin` service is provisioned when a channel called `default-admin` is added to domain custom resource definition and `AdministrationPortEnabled` is set to true in WebLogic Configuration

2. Verifies the `default-secure` service is provisioned when  a channel called `default-secure` is added to domain custom resource definition and SSL is enabled in WebLogic configuration.

(1) https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1997/  ( all tests passed ) 
(2) https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2090  ( with code refactoring)

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+  @DisplayName("Verify the secure service through administration port")
+  public void testVerifyAdminPortService() {
+    int sslNodePort = getServiceNodePort(
+         domainNamespace, adminServerPodName + "-external", "default-admin");
+    logger.info("Found the administration service nodePort {0}", sslNodePort);
+    String curlCmd = "curl -sk --show-error --noproxy '*' "
+        + " https://" + K8S_NODEPORT_HOST + ":" + sslNodePort
+        + "/console/login/LoginForm.jsp --write-out %{http_code} -o /dev/null";
+    logger.info("Executing defaukt-admin nodeport curl command {0}", curlCmd);
+    assertTrue(callWebAppAndWaitTillReady(curlCmd, 10));
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java:208]: You can use this dynamic wait if you really want to wait till ready.

    withStandardRetryPolicy
        .conditionEvaluationListener(
            condition -> logger.info("Waiting for clusterview app in admin server is accessible after restart"
                + "(elapsed time {0} ms, remaining time {1} ms)",
                condition.getElapsedTimeInMS(),
                condition.getRemainingTimeInMS()))
        .until((Callable<Boolean>) () -> {
          HttpResponse<String> response = assertDoesNotThrow(() -> OracleHttpClient.get(baseUri + serverListUri, true));
          return response.statusCode() == 200;
        });

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    logger.info("WebLogic console is accessible thru default-admin service");
+
+    int nodePort = getServiceNodePort(
+           domainNamespace, adminServerPodName + "-external", "default");
+    logger.info("Found the default service nodePort {0}", nodePort);
+    curlCmd = "curl -s --show-error --noproxy '*' "
+        + " http://" + K8S_NODEPORT_HOST + ":" + nodePort
+        + "/console/login/LoginForm.jsp --write-out %{http_code} -o /dev/null";
+    logger.info("Executing default nodeport curl command {0}", curlCmd);
+    assertFalse(callWebAppAndWaitTillReady(curlCmd, 5));
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java:218]: same as above comment

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    logger.info("WebLogic console is not accessible thru default service");
+  }
+
+  private static void createDomainSecret(String secretName, String username, String password, String domNamespace)
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java:222]: why not use the createSecretWithUsernamePassword from CommonTestUtils?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+    logger.info("Create domain custom resource for domainUid {0} in namespace {1}",
+            domainUid, domNamespace);
+    boolean domCreated = assertDoesNotThrow(() -> createDomainCustomResource(domain),
+            String.format("Create domain custom resource failed with ApiException for %s in namespace %s",
+                    domainUid, domNamespace));
+    assertTrue(domCreated, String.format("Create domain custom resource failed with ApiException "
+                    + "for %s in namespace %s", domainUid, domNamespace));
+  }
+
+  // Crate a ConfigMap with a model to add a 2 independent managed servers
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java:293]: update comment wrt this testcase

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * This test class verifies the `default-secure` service is provisioned when 
+ * a channel called `default-secure` is added to domain custom resource 
+ * definition and SSL is enabled in WebLogic configuration.
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:67]:
```suggestion
 * and SSL is enabled in WebLogic configuration.
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+ */
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:70]:
```suggestion
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+@DisplayName("Test secure nodePort service through SSL port and default-secure channel in domain CRD in a mii domain")
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:71]:
```suggestion
@DisplayName("Test secure nodePort service through SSL port and default-secure channel in domain CR in a mii domain")
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+  @BeforeAll
+  public static void initAll(@Namespaces(2) List<String> namespaces) {
+    logger = getLogger();
+    // create standard, reusable retry/backoff policy
+    withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+        .and().with().pollInterval(10, SECONDS)
+        .atMost(5, MINUTES).await();
+
+    // get a new unique opNamespace
+    logger.info("Creating unique namespace for Operator");
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:108]:
```suggestion
    logger.info("Assigning unique namespace for Operator");
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    logger.info("Creating unique namespace for Domain");
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:112]:
```suggestion
    logger.info("Assigning unique namespace for Domain");
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // Create the repo secret to pull the image
+    assertDoesNotThrow(() -> createDockerRegistrySecret(domainNamespace),
+          String.format("createSecret failed for %s", REPO_SECRET_NAME));
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    String adminSecretName = "weblogic-credentials";
+    assertDoesNotThrow(() -> createDomainSecret(adminSecretName,"weblogic",
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:126]:
```suggestion
    assertDoesNotThrow(() -> createDomainSecret(adminSecretName, ADMIN_USERNAME_DEFAULT,
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+            "welcome1", domainNamespace),
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:127]:
```suggestion
            ADMIN_PASSWORD_DEFAULT , domainNamespace),
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 12]
```diff
@@ -0,0 +1,312 @@
...
+             String.format("createSecret failed for %s", encryptionSecretName));
+
+    String configMapName = "default-admin-configmap";
+    String yamlString = "topology:\n"
+        + "  AdministrationPortEnabled: true\n"
+        + "  AdministrationPort: '9010'\n";
+
+    createModelConfigMap(configMapName, yamlString);
+
+    //determine if the tests are running in Kind cluster. if true use images from Kind registry
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java:145]: where are you using this wlsBaseImage  image?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+    String configMapName = "default-secure-configmap";
+    String yamlString = "topology:\n"
+        + "  Server:\n"
+        + "    'admin-server':\n"
+        + "       SSL: \n"
+        + "         Enabled: true \n"
+        + "         ListenPort: '7008' \n";
+    createModelConfigMap(configMapName, yamlString);
+
+    //determine if the tests are running in Kind cluster. if true use images from Kind registry
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:146]: where are you using this wlsBaseImage image?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 14]
```diff
@@ -0,0 +1,313 @@
...
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server services and pods are created in namespace {0}",
+          domainNamespace);
+      checkPodReadyAndServiceExists(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+  }
+
+  /**
+   * Create a WebLogic domain with SSL enabled in WebLogic configuration.
+   * Create a domain CRD with a channel with the name `default-secure`.
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:194]:
```suggestion
   * Create a domain CR with a channel with the name `default-secure`.
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 15]
```diff
@@ -0,0 +1,313 @@
...
+  @DisplayName("Verify the secure service through SSL port")
+  public void testVerifySecurePortService() {
+    int sslNodePort = getServiceNodePort(
+         domainNamespace, adminServerPodName + "-external", "default-secure");
+    logger.info("Found the administration service nodePort {0}", sslNodePort);
+    String curlCmd = "curl -sk --show-error --noproxy '*' "
+        + " https://" + K8S_NODEPORT_HOST + ":" + sslNodePort
+        + "/console/login/LoginForm.jsp --write-out %{http_code} -o /dev/null";
+    logger.info("Executing defaukt-admin nodeport curl command {0}", curlCmd);
+    assertTrue(callWebAppAndWaitTillReady(curlCmd, 10));
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:209]: same comment as the other test class

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 16]
```diff
@@ -0,0 +1,313 @@
...
+    logger.info("WebLogic console is accessible thru default-secure service");
+
+    int nodePort = getServiceNodePort(
+           domainNamespace, adminServerPodName + "-external", "default");
+    logger.info("Found the default service nodePort {0}", nodePort);
+    curlCmd = "curl -s --show-error --noproxy '*' "
+        + " http://" + K8S_NODEPORT_HOST + ":" + nodePort
+        + "/console/login/LoginForm.jsp --write-out %{http_code} -o /dev/null";
+    logger.info("Executing default nodeport curl command {0}", curlCmd);
+    assertTrue(callWebAppAndWaitTillReady(curlCmd, 5));
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:219]: same comment as the other test class

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 17]
```diff
@@ -0,0 +1,313 @@
...
+    logger.info("WebLogic console is accessible thru default service");
+  }
+
+  private static void createDomainSecret(String secretName, String username, String password, String domNamespace)
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:223]: same comment as the other test class

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 18]
```diff
@@ -0,0 +1,313 @@
...
+    logger.info("Create domain custom resource for domainUid {0} in namespace {1}",
+            domainUid, domNamespace);
+    boolean domCreated = assertDoesNotThrow(() -> createDomainCustomResource(domain),
+            String.format("Create domain custom resource failed with ApiException for %s in namespace %s",
+                    domainUid, domNamespace));
+    assertTrue(domCreated, String.format("Create domain custom resource failed with ApiException "
+                    + "for %s in namespace %s", domainUid, domNamespace));
+  }
+
+  // Crate a ConfigMap with a model to add a 2 independent managed servers
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:294]: same comment as the other test class

**Review by sankarpn — CHANGES_REQUESTED** [review]

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 19]
```diff
@@ -0,0 +1,312 @@
...
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * This test class verifies the `default-admin` service is provisioned when 
+ * a channel called `default-admin` is added to domain custom resource 
+ * definition and `AdministrationPortEnabled` is set to true.
+ */
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@DisplayName("Test secure nodePort service through admin port and default-admin channel in domain CRD in a mii domain")
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: Modified  
@DisplayName(Test secure nodePort service through admin port and default-admin channel in a mii domain")

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 20]
```diff
@@ -0,0 +1,312 @@
...
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * This test class verifies the `default-admin` service is provisioned when 
+ * a channel called `default-admin` is added to domain custom resource 
+ * definition and `AdministrationPortEnabled` is set to true.
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 21]
```diff
@@ -0,0 +1,312 @@
...
+
+  private final String adminServerPodName = domainUid + "-admin-server";
+  private final String managedServerPrefix = domainUid + "-managed-server";
+  private final String adminServerName = "admin-server";
+
+  private static LoggingFacade logger = null;
+
+  /**
+   * Install Operator.
+   * Create domain resource definition.
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 22]
```diff
@@ -0,0 +1,312 @@
...
+  @BeforeAll
+  public static void initAll(@Namespaces(2) List<String> namespaces) {
+    logger = getLogger();
+    // create standard, reusable retry/backoff policy
+    withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+        .and().with().pollInterval(10, SECONDS)
+        .atMost(5, MINUTES).await();
+
+    // get a new unique opNamespace
+    logger.info("Creating unique namespace for Operator");
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: Replaced Creating ->Assigning

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 14 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 16 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:218]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 17 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:222]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 18 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:293]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 23]
```diff
@@ -0,0 +1,312 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * This test class verifies the `default-admin` service is provisioned when 
+ * a channel called `default-admin` is added to domain custom resource 
+ * definition and `AdministrationPortEnabled` is set to true.
+ */
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: removed the empty line

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 24]
```diff
@@ -0,0 +1,312 @@
...
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // Create the repo secret to pull the image
+    assertDoesNotThrow(() -> createDockerRegistrySecret(domainNamespace),
+          String.format("createSecret failed for %s", REPO_SECRET_NAME));
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    String adminSecretName = "weblogic-credentials";
+    assertDoesNotThrow(() -> createDomainSecret(adminSecretName,"weblogic",
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: modified weblogic--> ADMIN_USERNAME_DEFAULT welcome --> ADMIN_PASSWORD_DEFAULT

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 25]
```diff
@@ -0,0 +1,312 @@
...
+            "welcome1", domainNamespace),
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: modified weblogic--> ADMIN_USERNAME_DEFAULT welcome --> ADMIN_PASSWORD_DEFAULT

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 26]
```diff
@@ -0,0 +1,312 @@
...
+  @Test
+  @DisplayName("Verify the secure service through administration port")
+  public void testVerifyAdminPortService() {
+    int sslNodePort = getServiceNodePort(
+         domainNamespace, adminServerPodName + "-external", "default-admin");
+    logger.info("Found the administration service nodePort {0}", sslNodePort);
+    String curlCmd = "curl -sk --show-error --noproxy '*' "
+        + " https://" + K8S_NODEPORT_HOST + ":" + sslNodePort
+        + "/console/login/LoginForm.jsp --write-out %{http_code} -o /dev/null";
+    logger.info("Executing defaukt-admin nodeport curl command {0}", curlCmd);
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 6 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 5 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: Modified as follows ....

This test class verifies the `default-secure` service is provisioned when 
a channel called `default-secure` is added to domain resource and  
SSL is enabled in WebLogic configuration.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java (hunk 4 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: Modified as follows ...
create a ConfigMap with a model file that set AdministrationPortEnabled to  true

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 7 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: @DisplayName("Test secure nodePort service through SSL port and default-secure channel in a mii domain")

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 9 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 10 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 11 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 8 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: done

**Comment by vanajamukkara** [comment]
@anpanigr can these use cases be added in existing test or domain by adding the config map configuration and channels in domain CR?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java (hunk 3 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: Modified test to use  CommonTestUtils method

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 13 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: These are boiler plate generic code from existing Mii Domain Class.  I am not building any domain images in this class. The ImageBuilder class will build the domain image by default.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 15 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: These are boiler plate generic code from existing Mii Domain Class.  I am not building any domain images in this class. The ImageBuilder class will build the domain image by default.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: Now I have added a check for the NodePort Service to make sure that, it finds a valid integer port.
 
    int nodePort = getServiceNodePort(
           domainNamespace, adminServerPodName + "-external", "default");
    assertTrue(nodePort != -1 ,  "Could not get the default external service node port");

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java (hunk 2 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: Now I have added a check for the NodePort Service to make sure that, it finds a valid integer port.

int nodePort = getServiceNodePort(
       domainNamespace, adminServerPodName + "-external", "default");
assertTrue(nodePort != -1 ,  "Could not get the default external service node port");

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 27]
```diff
@@ -0,0 +1,301 @@
...
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * This test class verifies the `default-secure` service is provisioned when 
+ * a channel called `default-secure` is added to domain resource and 
+ * SSL is enabled in WebLogic configuration.
+ */
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:69]: There is only one test method in this class, why do you need TestMethodOrder?

**Review by sankarpn — CHANGES_REQUESTED** [review]

**Comment by vanajamukkara** [comment]
> @anpanigr can these use cases be added in existing test or domain by adding the config map configuration and channels in domain CR?
> 
> @vanajamukkara We can not enable SSL and AdministartionPort in the same domain even if we create two channels. Those are mutually exclusive.

How about testing them in two different existing domains rather than creating new domains?

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 28]
```diff
@@ -0,0 +1,281 @@
...
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * This test class verifies the `default-admin` service is provisioned when 
+ * a channel called `default-admin` is added to domain resource and 
+ * `AdministrationPortEnabled` is set to true.
+ */
+@DisplayName("Test secure nodePort service through admin port and default-admin channel in a mii domain")
+@IntegrationTest
+class ItSecureAdminNodePort {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java:66]: as we discussed please change the It class name

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java [hunk 29]
```diff
@@ -0,0 +1,301 @@
...
+@DisplayName("Test secure nodePort service through SSL port and default-secure channel in a mii domain")
+@IntegrationTest
+class ItSecureNodePort {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java:72]: as discussed move this test to an existing mii domain

**Review by vanajamukkara — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 30]
```diff
@@ -0,0 +1,311 @@
...
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * This test class verifies the `default-admin` service is provisioned when 
+ * a channel called `default-admin` is added to domain resource and 
+ * `AdministrationPortEnabled` is set to true.
+ */
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java [hunk 31]
```diff
@@ -0,0 +1,311 @@
...
+
+    // Create the repo secret to pull the image
+    assertDoesNotThrow(() -> createDockerRegistrySecret(domainNamespace),
+          String.format("createSecret failed for %s", REPO_SECRET_NAME));
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    String adminSecretName = "weblogic-credentials";
+    assertDoesNotThrow(() -> createDomainSecret(adminSecretName,"weblogic",
+            "welcome1", domainNamespace),
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 27 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java (hunk 28 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureAdminNodePort.java]: Now the class name is ItDefaultAdminNodePort

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java (hunk 29 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java]: Moved the test into ItMiiDomain.java and merger the method  testCreateMiiDomain()

**Review by vanajamukkara — APPROVED** [review]
LGTM

**Review by sankarpn — APPROVED** [review]
LGTM

**Pull request merged**